src/Uno.UI.Runtime.Skia.Win32/Win32WindowWrapper.Pointers.cs

```diff
@@ -204,7 +204,8 @@ internal unsafe void OnPointer(uint msg, WPARAM wParam, HWND hwnd)
 					properties.IsMiddleButtonPressed = Win32Helper.IS_POINTER_THIRDBUTTON_WPARAM(wParam);
 					properties.IsXButton1Pressed = Win32Helper.IS_POINTER_FOURTHBUTTON_WPARAM(wParam);
 					properties.IsXButton2Pressed = Win32Helper.IS_POINTER_FIFTHBUTTON_WPARAM(wParam);
-					properties.IsHorizontalMouseWheel = (wParam & (ulong)POINTER_FLAGS.POINTER_FLAG_HWHEEL) != 0;
+					// POINTER_FLAG_HWHEEL is set when mouse-scrolling with Shift held. TouchPad horizontal scrolling uses WM_POINTERHWHEEL and does not set POINTER_FLAG_HWHEEL
+					properties.IsHorizontalMouseWheel = msg is PInvoke.WM_POINTERHWHEEL || (wParam & (ulong)POINTER_FLAGS.POINTER_FLAG_HWHEEL) != 0;
```

Comment:
"POINTER_FLAG_HWHEEL is set when mouse-scrolling with Shift held"

Then we should NOT consider this as `Horizontal` scroll, but scroll with the `Shift` key modifier https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.pointereventargs.keymodifiers?view=windows-app-sdk-1.7

```suggestion
					properties.IsHorizontalMouseWheel = msg is PInvoke.WM_POINTERHWHEEL;
```

And then set the `point.KeyModifiers` property.

Reply:
BTW it would be nice to also add support of horizontal mouse wheel here https://github.com/unoplatform/uno/blob/61f760887b67ac9ebc440eb62691005163f8fcfe/src/Uno.UI.Runtime.Skia.Win32/Win32WindowWrapper.Pointers.cs#L145